dlib/media/ffmpeg_abstract.h

```diff
@@ -1025,33 +1025,25 @@ namespace dlib
           class Allocator,
           std::enable_if_t<is_byte<Byte>::value, bool> = true
         >
-        DEDUCED sink(std::vector<Byte, Allocator>& buf);
+        auto sink(std::vector<Byte, Allocator>& buf);
         /*!
             requires
                 - Byte must be a byte type, e.g. char, int8_t or uint8_t
-                - The lifetime of buf must exceed the lifetime of the encoder 
-                  instance you are passing this to.
-
             ensures
-                - Creates a "view" type which type-erases &buf and creates
-                  a suitable callback with signature bool(std::size_t, const char*)
-                  which can be passed to encoder.
-                - When the callback is invoked, packet data is appended to buf.
+                - returns a function object with signature bool(std::size_t N, const char* data).  When
+                  called that function appends the first N bytes pointed to by data onto the end of buf.
+                - The returned function is valid only as long as buf exists.
+                - The function always returns true.        
         !*/
 
 // ---------------------------------------------------------------------------------------------------
 
-        DEDUCED sink(std::ostream& out);
+        auto sink(std::ostream& out);
         /*!
-            requires
-                - The lifetime of out must exceed the lifetime of the encoder 
-                  instance you are passing this to.
-
             ensures
-                - Creates a "view" type which type-erases &out and creates
-                  a suitable callback with signature bool(std::size_t, const char*)
-                  which can be passed to encoder.
-                - When the callback is invoked, packet data is added to the output stream.
+                - returns a function object with signature bool(std::size_t N, const char* data).  When
+                  called that function writes the first N bytes pointed to by data to out.
+                - The returned view is valid only as long as out exists.                 
```

Comment:
```suggestion
                - The returned view is valid only as long as out exists.
                - Returns out.good(). I.e. returns true if the write to the stream succeeded and false otherwise.       
```